Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -5812,5 +5821,44 @@ RefPtr<CSSValue> consumeViewTimelineInset(CSSParserTokenRange& range, const CSSP
     });
 }
 
+RefPtr<CSSPrimitiveValue> consumeAnchor(CSSParserTokenRange& range, CSSParserMode mode)
+{
+    // https://drafts.csswg.org/css-anchor-position-1/#anchor-pos
+    // <anchor()> = anchor( <anchor-element>? && <anchor-side>, <length-percentage>? )
+    auto rangeCopy = range;
+
+    if (rangeCopy.peek().type() != FunctionToken || range.peek().functionId() != CSSValueAnchor)
+        return nullptr;
+
+    auto args = consumeFunction(rangeCopy);
+    if (!args.size())
+        return nullptr;
+
+    auto anchorElement = consumeDashedIdent(args);
+    auto anchorSide = CSSPropertyParsing::consumeAnchorSide(args);
+    if (!anchorSide)
+        return nullptr;
+
+    if (!anchorElement)
+        anchorElement = consumeDashedIdent(args);
+
+    if (!args.size()) {
+        range = rangeCopy;
+        auto anchor = CSSAnchorValue::create(WTFMove(anchorElement), WTFMove(anchorSide), nullptr);
+        return CSSPrimitiveValue::create(anchor);
```

Comment:
I don’t get why we wrap `CSSAnchorValue` inside a `CSSPrimitiveValue`.

Reply:
I suggested just following `calc()` pattern as deviating from it will likely result in a larger refactoring.

Reply:
That sort of cleanup should also be done for `calc()` and might be better done separately for both.